### ISSUE TITLE
Plugin does not detect that keyword can not be used as lifetime name

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#8046]
## Environment

* **IntelliJ Rust plugin version:** 0.4.160.4218-213-nightly
* **Rust toolchain version:** 1.56.0 (09c42c458 2021-10-18) x86_64-apple-darwin
* **IDE name and version:** CLion 2021.3 EAP (CL-213.5449.26)
* **Operating system:** macOS 11.4
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description
Error "lifetimes cannot use keyword names" does not detected.

## Steps to reproduce
```rust
//keyword `let`:
fn bar(let: i32) { 

}
//keyword `let`:
fn foo<'let, 'b>(c: &'let i32, d: &'b i32) -> &'let i32 {
    c
}
```

## Actual result 
<img width="752" alt="image" src="https://user-images.githubusercontent.com/49211026/139846024-47df910b-df8b-4286-b2a3-747f8ee21373.png">

## Expected result 
(with Cargo Check)
<img width="750" alt="image" src="https://user-images.githubusercontent.com/49211026/139844948-2c2e1a40-6149-4885-acec-258561458945.png">

**Comment by Kobzol** [comment]
Is this different from https://github.com/intellij-rust/intellij-rust/issues/6927? It was fixed in https://github.com/intellij-rust/intellij-rust/pull/6929.

**Comment by neonaot** [comment]
No different, I forgot to look in the closed issues :D
Fixed

**Comment by Undin** [comment]
Duplicate of #6927
